frontend/packages/console-dynamic-plugin-sdk/src/app/k8s/api-discovery/api-discovery.ts

```diff
@@ -0,0 +1,174 @@
+  try {
+    apiResourceData = await consoleFetchJSON(`${k8sBasePath}/apis`);
+  } catch (err) {
+    throw err;
+  }
+  const groupVersionMap = apiResourceData.groups.reduce(
+    (acc, { name, versions, preferredVersion: { version } }) => {
+      acc[name] = {
+        versions: _.map(versions, 'version'),
+        preferredVersion: version,
+      };
+      return acc;
+    },
+    {},
+  );
+  const all: Promise<APIResourceList>[] = _.flatten(
+    apiResourceData.groups.map((group) =>
+      group.versions.map((version) => `/apis/${version.groupVersion}`),
+    ),
+  )
+    .concat(['/api/v1'])
+    .map((p) => consoleFetchJSON(`api/kubernetes${p}`).catch((err) => err));
+
+  return Promise.all(all).then((data) => {
+    const resourceSet = new Set<string>();
+    const namespacedSet = new Set<string>();
+    data.forEach(
+      (d) =>
+        d.resources &&
+        d.resources.forEach(({ namespaced, name }) => {
+          resourceSet.add(name);
+          namespaced && namespacedSet.add(name);
+        }),
+    );
+    const allResources = [...resourceSet].sort();
+
+    const safeResources = [];
+    const adminResources = [];
+    const models = _.flatten(data.filter((d) => d.resources).map(defineModels));
+    const coreResources = new Set([
+      'roles',
+      'rolebindings',
+      'clusterroles',
+      'clusterrolebindings',
+      'thirdpartyresources',
+      'nodes',
+      'secrets',
+    ]);
+    allResources.forEach((r) =>
+      coreResources.has(r.split('/')[0]) ? adminResources.push(r) : safeResources.push(r),
+    );
+    const configResources = _.filter(
+      models,
+      (m) => m.apiGroup === 'config.openshift.io' && m.kind !== 'ClusterOperator',
+    );
+    const clusterOperatorConfigResources = _.filter(
+      models,
+      (m) => m.apiGroup === 'operator.openshift.io',
+    );
+
+    return {
+      allResources,
+      safeResources,
+      adminResources,
+      configResources,
+      clusterOperatorConfigResources,
+      namespacedSet,
+      models,
+      groupVersionMap,
+    } as DiscoveryResources;
+  });
+};
+
+const updateResources = () => (dispatch: Dispatch) => {
+  dispatch(getResourcesInFlight());
+
+  getResources()
+    .then((resources) => {
+      // Cache the resources whenever discovery completes to improve console load times.
+      cacheResources(resources);
+      dispatch(receivedResources(resources));
+    })
+    // eslint-disable-next-line no-console
+    .catch((err) => console.error('Fetching resource failed:', err));
+};
+
+const startAPIDiscoverySDK = () => (dispatch) => {
+  // eslint-disable-next-line no-console
+  console.log('API discovery method: Polling');
+  // Poll API discovery every 30 seconds since we can't watch CRDs
+  dispatch(updateResources());
+  if (POLLs[apiDiscovery]) {
+    clearTimeout(POLLs[apiDiscovery]);
+    delete POLLs[apiDiscovery];
+  }
+  POLLs[apiDiscovery] = setTimeout(() => dispatch(startAPIDiscoverySDK()), 60000);
```

Comment:
Nit: can we store `60000` in a const at the top of the file so we can find it easier later?
`const API_DISCOVERY_POLL_INTERVAL = 60000;`